endpoints/openrtb2/auction.go

```diff
@@ -50,8 +58,16 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 		return
 	}
 
-	usersyncs := pbs.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie.OptOutCookie))
+	ctx := context.Background()
```

Comment:
Is there something in stored bid requests that necessitated moving the context generation up out of parseRequest(), or was it just an extra fix that snuck in along with this PR?

In either case I think the context does better belong in the Auction.

Reply:
I think it's a little of both.

Our old code assumed that the Timeout was on the BidRequest. I know I had to rewrite it partially because that Timeout may now be in the Stored BidRequest data. There's a catch 22. We need a `context` to fetch the Stored Request data, but the timeout may not be known until we _have_ that data.

Once I rewrite a block of code, if I have time I'll usually step back and rewrite it 3-4 more times to make it as simple as I can. I think my first working version added a 5th return value to `parseRequest`, or something like that, which made the code "too ugly" to pass through the "simplify" phase.